### PR TITLE
lightning(dm): use working directory for sort-kv when source-dir is S3

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #3510

### What is changed and how it works?

as title

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/config/subtask.go [hunk 1]
```diff
@@ -333,6 +334,13 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 			return terror.ErrConfigLoaderDirInvalid.Delegate(err, c.LoaderConfig.Dir)
 		}
 		c.LoaderConfig.Dir = newDir
+
+		if storage.IsLocalDiskPath(newDir) {
+			// lightning will not recursively create directories, so we use same level dir
+			c.LoaderConfig.SortingDirPhysical = newDir + ".sorting"
```
Comment (gozssky) [inline dm/config/subtask.go:340]: Is `newDir` the data source directory? Does it affect the mydump read process?

Comment (lance6716) [reply dm/config/subtask.go:340]: yes. `./dump_data` and `./dump_data.sorting` are two directories, it should not be affected

**Review by buchuitoudegou — APPROVED** [review]

**Pull request merged**